line_profiler/_map_helpers.pxi

```diff
@@ -0,0 +1,64 @@
+# cython: language_level=3
+# cython: legacy_implicit_noexcept=True
+# used in _line_profiler.pyx
+from libcpp.unordered_map cimport unordered_map
+from cython.operator cimport dereference as deref
+
+# If your module already defines PY_LONG_LONG, reuse that.
+# long long int is at least 64 bytes assuming c99
+ctypedef unsigned long long int uint64
+ctypedef long long int int64
+
+cdef extern from "Python_wrapper.h":
+    ctypedef long long PY_LONG_LONG
+
+cdef struct LastTime:
+    int f_lineno
+    PY_LONG_LONG time
+
+cdef struct LineTime:
+    long long code
+    int lineno
+    PY_LONG_LONG total_time
+    long nhits
+
+# Types used for mappings from code hash to last/line times.
+ctypedef unordered_map[int64, LastTime] LastTimeMap
+ctypedef unordered_map[int64, LineTime] LineTimeMap
+
+cdef extern from "timers.c":
+    PY_LONG_LONG hpTimer()  # add 'nogil' here only if it truly is nogil
+
+# ---- helpers ----
+
+cdef inline bint last_contains(LastTimeMap* m, int64 key) noexcept nogil:
```

Comment:
Marking a bunch of short helpers nogil tends to hurt performance because the cost of acquiring and releasing the GIL when we switch from a function with python interaction to one without usually outweighs any potential multicore benefits. Did you benchmark these changes to be sure?